corehq/form_processor/submission_post.py

```diff
@@ -409,12 +410,16 @@ def _notify_submission_error(instance, exception, message):
 
 @contextlib.contextmanager
 def unfinished_submission(instance):
-    unfinished_submission_stub = UnfinishedSubmissionStub.objects.create(
-        xform_id=instance.form_id,
-        timestamp=datetime.datetime.utcnow(),
-        saved=False,
-        domain=instance.domain,
-    )
+    unfinished_submission_stub = None
+    if not getattr(instance, 'deprecated_form_id', None):
+        # don't create stubs for form edits
+        unfinished_submission_stub = UnfinishedSubmissionStub.objects.create(
+            xform_id=instance.form_id,
+            timestamp=datetime.datetime.utcnow(),
+            saved=False,
+            domain=instance.domain,
+        )
     yield unfinished_submission_stub
 
-    unfinished_submission_stub.delete()
+    if unfinished_submission_stub:
+        unfinished_submission_stub.delete()
```

Comment:
should this stub be deleted if the context manager exits with an error? if yes, should use `try`/`finally` around the `yield`.

Reply:
no, if it exits with an error then it should not be deleted, that's most of the point of the it.